src/olympia/addons/decorators.py

```diff
@@ -55,3 +61,30 @@ def addon_view_factory(qs):
     # GOOD: lambda: Addon.objects.valid().filter(type=1)
     # BAD: Addon.objects.valid()
     return functools.partial(addon_view, qs=qs)
+
+
+def require_submissions_enabled(f):
+    """Require the enable-submissions waffle flag to be enabled."""
+
+    @functools.wraps(f)
+    def wrapper(request, *args, **kw):
+        if waffle.flag_is_active(request, 'enable-submissions'):
+            return f(request, *args, **kw)
+        flag = waffle.get_waffle_flag_model().get('enable-submissions')
+        reason = flag.note if hasattr(flag, 'note') else None
+        if getattr(request, 'is_api', True):
+            return Response(
+                {
+                    'error': gettext('Add-on uploads are temporarily unavailable.'),
+                    'reason': reason,
+                },
+                status=status.HTTP_403_FORBIDDEN,
```

Comment:
Here and below: I think we should use a 503, not a 403. It's a temporary error caused by us, not something the developer can do anything about.